src/core/coap/coap.cpp

```diff
@@ -962,6 +962,37 @@ uint32_t ResponsesQueue::ResponseMetadata::GetRemainingTime(void) const
     return (mDequeueTime > now) ? mDequeueTime - now : 0;
 }
 
+/// Return product of @p aValueA and @p aValueB if no overflow otherwise 0.
+static uint32_t Multiple(uint32_t aValueA, uint32_t aValueB)
```

Comment:
Should we make this function a member of `TxParameters` since it has special rules that only applies to `TxParameters`?

Reply:
I prefer keep implementation in .cpp file if possible.